ttbd/ui/templates/widget-target-tunnels.html

```diff
@@ -0,0 +1,71 @@
+<!-- this is start of file ui/widget-target-tunnels.html -->
+
+<div class='status'>
+  {% if tunnels is none %}
+    <h4 class='info info-grey'>network tunnels: not supported</h4>
+  {% else %}
+    <h4 class='info info-grey'>
+      <div class = "tooltip">
+        network tunnels
+        <span class = "tooltiptext">
+          Tunnel from the server into private networks to which
+          both the server and targets are connected.
+        </span>
+      </div>
+    </h4>
+    <table class='display'>
+      <thead>
+        <th align="left">Protocol</th>
+        <th align="left">From</th>
+        <th align="left">To</th>
+        <th align="left">Action</th>
+      </thead>
+
+      <!-- tunnels is [ ( protocol, ip_addr, port, local_port )] -->
+      {% for protocol, ip_addr, port, local_port in tunnels %}
+      <tr>
+        <td>{{ protocol }} </td>
+        <td>
+          <div class = "tooltip">
+            {{ local_port }}…
```

Comment:
what does the `...` represent?

Reply:
indicating there is more info in the tooltip -- namely the details of the IP addresses they can use to connect to it